### ISSUE TITLE
error message for missing main function in example is non-specific

### ISSUE THREAD

**Issue by huonw** [rust-lang/cargo#1361]
```
$ cargo new missing-main
$ cd missing-main
$ mkdir examples
$ touch examples/foo.rs
$ cargo build
   Compiling missing-main v0.0.1 (file:///home/huon/tmp/missing-main)
error: main function not found
error: aborting due to previous error
Could not compile `missing-main`.
```

Running with `--verbose` does at least print the last command run (pointing the finger at `examples/foo,rs`), but it seems bad to have a raw `error:` printed like this.

**Comment by alexcrichton** [comment]
Hm perhaps this is more of a compiler issue than a Cargo issue? Cargo currently does not interpret errors at all, it just ships them back to the front. Although one possible step Cargo could take is:

```
Could not compile `missing-main` at `examples/foo.rs`
```

(which seems plausible?)

**Comment by carols10cents** [comment]
Triage: this is still a problem, but I had to either `cargo run --example foo` or `cargo test`; looks like `cargo build` no longer builds examples by default.

The error messages still don't mention `examples/foo.rs` though, which is the problem.

```
$ cargo new missing-main
     Created library `missing-main` project
$ cd missing-main
$ mkdir examples
$ touch examples/foo.rs
$ cargo build
   Compiling missing-main v0.1.0 (file:///Users/carolnichols/rust/missing-main)
    Finished dev [unoptimized + debuginfo] target(s) in 0.19 secs

$ cargo build --verbose
       Fresh missing-main v0.1.0 (file:///Users/carolnichols/rust/missing-main)
    Finished dev [unoptimized + debuginfo] target(s) in 0.0 secs

$ cargo run --example foo
   Compiling missing-main v0.1.0 (file:///Users/carolnichols/rust/missing-main)
error: main function not found

error: aborting due to previous error

error: Could not compile `missing-main`.

To learn more, run the command again with --verbose.

$ cargo run --example foo --verbose
   Compiling missing-main v0.1.0 (file:///Users/carolnichols/rust/missing-main)
     Running `rustc --crate-name foo examples/foo.rs --crate-type bin --emit=dep-info,link -C debuginfo=2 -C metadata=799ee0edb02a8799 -C extra-filename=-799ee0edb02a8799 --out-dir /Users/carolnichols/rust/missing-main/target/debug/examples -L dependency=/Users/carolnichols/rust/missing-main/target/debug/deps --extern missing_main=/Users/carolnichols/rust/missing-main/target/debug/deps/libmissing_main-4976ac668ccc2721.rlib`
error: main function not found

error: aborting due to previous error

error: Could not compile `missing-main`.

Caused by:
  process didn't exit successfully: `rustc --crate-name foo examples/foo.rs --crate-type bin --emit=dep-info,link -C debuginfo=2 -C metadata=799ee0edb02a8799 -C extra-filename=-799ee0edb02a8799 --out-dir /Users/carolnichols/rust/missing-main/target/debug/examples -L dependency=/Users/carolnichols/rust/missing-main/target/debug/deps --extern missing_main=/Users/carolnichols/rust/missing-main/target/debug/deps/libmissing_main-4976ac668ccc2721.rlib` (exit code: 101)

$ cargo test
   Compiling missing-main v0.1.0 (file:///Users/carolnichols/rust/missing-main)
error: main function not found

error: aborting due to previous error

error: Could not compile `missing-main`.
Build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by ehuss** [comment]
I think this is resolved by rust-lang/rust#48706.
